tests/nat/nat_helpers.py

```diff
@@ -0,0 +1,848 @@
+    :param interface_type: type of interface
+    :param second_port: boolean if second port's IP settings need to be returned
+    :return: public IP
+    """
+    if second_port and interface_type == "port_in_lag":
+        return setup_info[interface_type]['second_public_ip']
+    return setup_info[interface_type]['public_ip']
+
+
+def setup_ptf_interfaces(ptfhost, duthost, setup_info, interface_type, vrf_id, vrf_name, port_id,
+                         ip_address, mask, gw_ip, key):
+    """
+    setup ptf interfaces for tests
+    :param ptfhost: PTF host object
+    :param duthost: DUT host object
+    :param setup_info: setup info fixture
+    :param interface_type: string interface type
+    :param vrf_id: id of vrf
+    :param vrf_name: vrf name
+    :param port_id: port id of interface
+    :param ip_address: ip address of interface
+    :param mask: vrf mask
+    :param gw_ip: ip address of gateway
+    :param key: dictionary key if vrf configuration
+    """
+    ptfhost.shell("grep -Fxq '{} {}' /etc/iproute2/rt_tables "
+                  "|| echo '{} {}' >> /etc/iproute2/rt_tables".format(vrf_id, vrf_name, vrf_id, vrf_name))
+    ptfhost.shell("ip link add {} type vrf table {}".format(vrf_name, vrf_id))
+    ptfhost.shell("ip link set dev {} up".format(vrf_name))
+    ptfhost.shell("ip link set eth{} master {}".format(port_id, vrf_name))
+    ptfhost.shell("ip rule add iif {} table {}".format(vrf_name, vrf_id))
+    ptfhost.shell("ip rule add oif {} table {}".format(vrf_name, vrf_id))
+    ptfhost.shell("ip addr add {}/{} dev eth{}".format(ip_address, mask, port_id))
+    ptfhost.shell("ip route add 0.0.0.0/0 via {} table {}".format(gw_ip, vrf_id))
+    if "dut_iface" in setup_info[interface_type]["vrf_conf"][key].keys():
+        dut_iface = setup_info[interface_type]["vrf_conf"][key]["dut_iface"]
+        pch_ip = setup_info["pch_ips"][dut_iface]
+        duthost.shell("sudo config interface ip remove {} {}/31".format(dut_iface, pch_ip))
+        duthost.shell("sudo config interface ip add {} {}/{}".format(dut_iface, gw_ip, mask))
+
+
+def teardown_ptf_interfaces(ptfhost, gw_ip, vrf_id, ip_address, mask, port_id, vrf_name):
+    """
+    teardown ptf interfaces after tests
+    :param ptfhost: PTF host object
+    :param vrf_id: id of vrf
+    :param vrf_name: vrf name
+    :param port_id: port id of interface
+    :param ip_address: ip address of interface
+    :param mask: vrf mask
+    :param gw_ip: ip address of gateway
+    """
+    ptfhost.shell("ip route del 0.0.0.0/0 via {} table {}".format(gw_ip, vrf_id))
+    ptfhost.shell("ip addr del {}/{} dev eth{}".format(ip_address, mask, port_id))
+    ptfhost.shell("ip rule del iif {} table {}".format(vrf_name, vrf_id))
+    ptfhost.shell("ip rule del oif {} table {}".format(vrf_name, vrf_id))
+    ptfhost.shell("ip link set eth{} nomaster".format(port_id))
+    ptfhost.shell("ip link del {} type vrf table {}".format(vrf_name, vrf_id))
+
+
```

Comment:
do we need to do the cleanup for this  ptfhost.shell("ip link add {} type vrf table {}".format(vrf_name, vrf_id))

Reply:
After end of the test session teardown procedure turns testbed to the initial state.